apps/dashboard/app/apps/ood_app.rb

```diff
@@ -308,7 +308,13 @@ def passenger_railsdb_app?
 
   # @return [String] memoized version string
   def version
-    @version ||= (version_from_file || version_from_git || "unknown").strip
+    #hide_app_version has to be the first entry
+    @version ||= (hide_app_version || version_from_file || version_from_git || "unknown").strip
```

Comment:
I think we should be passing `nil` to the view for cases where we don't want to display the version.  Passing strings keyword like `unknown` is bad practice. 

```suggestion
        @version ||= if Configuration.Configuration.hide_app_version?
                   nil
                 elsif version_from_file.to_s.downcase == 'unknown'
                   nil
                 else
                   (file_version || version_from_git).strip
                 end
```

For the view then, it either has something (not `nil`) or it doesn't (`nil`), so it doesn't need to check for string comparisons.

So this 
https://github.com/OSC/ondemand/blob/13eceeaed30a68bcea00053643ee60de2b73981c/apps/dashboard/app/views/batch_connect/session_contexts/new.html.erb#L78-L80

Can turn into a much simpler check for just nil.  I.e., you either have it or you don't, but the view doesn't retain knowledge about this `unknown` string feature.
```erb
      <% unless @app.version.nil? %>
        <small>version: <%= @app.version %></small>
      <% end %>
```

Reply:
@johrstrom   Your feedback is right on. However, we will need to ensure that there are no lingering dependencies on 'unknown'.  That would be my only concern.

Reply:
> You're thinking there's other things that rely on the keyword. A grep through the codebase doesn't seem to indicate such. Interestingly enough the product pages don't show the version.

Not necessarily thinking that there is other things depending on the keyword. I was just wanting to ensure that developer did what you did and ensured by a grep or some other means that there was no other dependency.